### PR TITLE
Consider platform API <= 0.6 and buildpack API <= 0.6 deprecated

### PR DESCRIPTION
**Pull request by natalieparellano**
According to https://github.com/buildpacks/rfcs/blob/main/text/0110-deprecate-apis.md

With a lifecycle packaged from this branch:

```
$  cd out && cat lifecycle.toml 
[apis]
[apis.buildpack]
  deprecated = ["0.2", "0.3", "0.4", "0.5", "0.6"]
  supported = ["0.2", "0.3", "0.4", "0.5", "0.6", "0.7", "0.8", "0.9"]
[apis.platform]
  deprecated = ["0.3", "0.4", "0.5", "0.6"]
  supported = ["0.3", "0.4", "0.5", "0.6", "0.7", "0.8", "0.9", "0.10", "0.11"]

# For backwards compatibility the minimum supported APIs are provided in RFC-0011 format
# https://github.com/buildpacks/rfcs/blob/main/text/0011-lifecycle-descriptor.md
[api]
  platform = "0.3"
  buildpack = "0.2"

[lifecycle]
  version = "0.15.1-7+02569f89"
  sbom-formats = ["application/vnd.cyclonedx+json", "application/spdx+json", "application/vnd.syft+json"]
```

```
$  docker run -it --rm -v $PWD:/lifecycle ubuntu:bionic                                      
root@ba16014118c7:/lifecycle# /lifecycle/lifecycle/lifecycle -version
0.15.1-7+02569f89
root@ba16014118c7:/lifecycle# CNB_PLATFORM_API=0.3 /lifecycle/lifecycle/lifecycle -version
Warning: Platform requested deprecated API '0.3'
0.15.1-7+02569f89
```

### CODE REVIEW

**Review by jabrown85 — APPROVED** [review]
I like this - a bit scary if folks aren't setting `CNB_PLATFORM_API` as the jump is huge. But they are likely on older lifecycle releases anyway I suppose...

**Comment by natalieparellano** [comment]
Hmm... maybe we shouldn't update the default platform API just yet. This will give people who are defaulting to the default a chance to see the warnings and upgrade. WDYT?

**Comment by jabrown85** [comment]
> Hmm... maybe we shouldn't update the default platform API just yet. This will give people who are defaulting to the default a chance to see the warnings and upgrade. WDYT?

I like that - but a fast follow in the next version seems reasonable

File: acceptance/launcher_test.go [hunk 1]
```diff
@@ -141,14 +142,14 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
 		when("there is no CMD provided", func() {
 			when("CNB_PROCESS_TYPE is NOT set", func() {
 				it("web is the default process-type", func() {
-					cmd := exec.Command("docker", "run", "--rm", launchImage)
+					cmd := exec.Command("docker", "run", "--rm", "--env=CNB_PLATFORM_API=0.3", launchImage)
```
[the remark this answers is not in the document]
Comment (jabrown85) [reply acceptance/launcher_test.go:145]: That makes sense to me

[the remark this answers is not in the document]
Comment (natalieparellano) [reply acceptance/launcher_test.go:146]: Done in e6979275

File: cmd/lifecycle/cli/command.go [hunk 2]
```diff
@@ -54,6 +55,13 @@ func Run(c Command, asSubcommand bool) {
 	if err := cmd.SetLogLevel(logLevel); err != nil {
 		cmd.Exit(err)
 	}
+
+	// Warn when CNB_PLATFORM_API is unset
```
Comment (natalieparellano) [inline cmd/lifecycle/cli/command.go:59]: With this placement we don't print the warning if we're just running `lifecycle -version`

**Comment by natalieparellano** [comment]
Sample output:
```
# /lifecycle/lifecycle analyze -version
Warning: Platform requested deprecated API '0.3'
0.15.1-12+e6979275

# /lifecycle/analyzer -version         
Warning: Platform requested deprecated API '0.3'
0.15.1-12+e6979275

# /lifecycle/lifecycle -version
0.15.1-12+e6979275
```

It's a bit annoying to see the warning in the first two cases, but the fix requires a bit more refactoring than I care to do now.

**Comment by natalieparellano** [comment]
@jabrown85 I think this is ready for another look

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**